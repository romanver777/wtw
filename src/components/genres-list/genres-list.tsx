import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import { MovieType } from "../../types/types";
import { GENRES_RUS_TO_ENG } from "../../helpers/const";

interface StateProps {
  films: MovieType[];
  activeGenre: string;
  genresList: string[];
}

interface MapProps {
  genresList: string[];
  activeGenre: string;
}

type Props = MapProps;

const GenresList = (props: Props): JSX.Element => {
  const { genresList, activeGenre } = props;

  return (
    <React.Fragment>
      {genresList && genresList.length && (
        <ul className="catalog__genres-list">
          {genresList.map((item, ind) => {
            return (
              <li
                className={
                  item === activeGenre
                    ? "catalog__genres-item catalog__genres-item--active"
                    : "catalog__genres-item"
                }
                key={ind}
              >
                <Link
                  to={`/?genre=${GENRES_RUS_TO_ENG[item]}`}
                  className="catalog__genres-link"
                  onClick={(e) => {
                    if (activeGenre === item) e.preventDefault();
                  }}
                >
                  {item}
                </Link>
              </li>
            );
          })}
        </ul>
      )}
    </React.Fragment>
  );
};

const mapStateToProps = (state: StateProps) => ({
  genresList: state.genresList,
  activeGenre: state.activeGenre,
});

export { GenresList };
export default connect(mapStateToProps)(GenresList);
