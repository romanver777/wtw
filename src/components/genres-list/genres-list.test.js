/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import { Router } from "react-router";
import history from "../../history";
import { GenresList } from "./genres-list";

describe("GenresList component", () => {
  const props = {
    genresList: ["Все жанры", "Комедия", "Фантастика"],
    activeGenre: "Комедия",
  };
  const Trenderer = TestRenderer.create(
    <Router history={history}>
      <GenresList {...props} />
    </Router>
  );

  it("renders correctly", () => {
    expect(Trenderer.toJSON()).toMatchSnapshot();
  });
  it("renders genres list", () => {
    const list = Trenderer.root.findAllByType("a").map((el) => el.children[0]);

    expect(list).toEqual(props.genresList);
  });
  it("renders active genre", () => {
    const className = "catalog__genres-item catalog__genres-item--active";

    expect(
      Trenderer.root.findByProps({ className: `${className}` }).props.children
        .props.children
    ).toEqual(props.activeGenre);
  });
});
