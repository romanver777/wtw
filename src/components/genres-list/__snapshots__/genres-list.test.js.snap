// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList component renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#/"
      onClick={[Function]}
    >
      Все жанры
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#/"
      onClick={[Function]}
    >
      Комедия
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#/"
      onClick={[Function]}
    >
      Фантастика
    </a>
  </li>
</ul>
`;
